### ISSUE TITLE
"can't open '/var/run/docker.pid': No such file or directory" on attempted daemon stop

### ISSUE THREAD

**Issue by garystafford** [docker/machine#1710]
On Ubuntu 15.04, with latest Machine, Docker, Compose, Swarm just installed:
```bash
Docker version 1.8.1, build d12ea79
docker-compose version: 1.4.0
docker-machine version 0.4.1 (e2c88d6)
swarm version 0.4.0 (d647d82)
vboxmanage 5.0.0r101573
```
Seeing even more ssh-related failures to creating VirtualBox VMs (swarm nodes).  Failed VMs ends up with same IP address as existing machine. Only fix is changing name of node and trying again. Not sure if same bugs that have been reported repeatedly with earlier versions of Machine:
```bash
gstafford-X555LA$ docker-machine create \
>   --driver virtualbox --virtualbox-disk-size 10000 \
>   --engine-label itype=frontend \
>   --swarm --swarm-discovery \
>   token://$scid swarm-node-01
Creating VirtualBox VM...
Creating SSH key...
Starting VirtualBox VM...
Starting VM...
SSH cmd error!
command: sudo /etc/init.d/docker stop
err    : exit status 1
output : cat: can't open '/var/run/docker.pid': No such file or directory
sh: you need to specify whom to kill
```

Note 01 and 02 IPs? Not expected behavior?
```bash
gstafford-X555LA$ docker-machine ls
NAME            ACTIVE   DRIVER       STATE     URL                         SWARM
swarm-master             virtualbox   Running   tcp://192.168.99.100:2376   swarm-master (master)
swarm-node-01            virtualbox   Running   tcp://192.168.99.101:2376   swarm-master
swarm-node-02            virtualbox   Running   tcp://192.168.99.101:2376   swarm-master
swarm-node-03            virtualbox   Running   tcp://192.168.99.102:2376   swarm-master
```
Remove node and try again, same command, new name
```bash
gstafford-X555LA$ docker-machine rm swarm-node-01 -f
Successfully removed swarm-node-01
gstafford-X555LA$ docker-machine create   --driver virtualbox --virtualbox-disk-size 10000   --engine-label itype=frontend   --swarm --swarm-discovery   token://$scid swarm-node-04
Creating VirtualBox VM...
Creating SSH key...
Starting VirtualBox VM...
Starting VM...
To see how to connect Docker to this machine, run: docker-machine env swarm-node-04

gstafford-X555LA$ docker-machine ls
NAME            ACTIVE   DRIVER       STATE     URL                         SWARM
swarm-master             virtualbox   Running   tcp://192.168.99.100:2376   swarm-master (master)
swarm-node-02            virtualbox   Running   tcp://192.168.99.101:2376   swarm-master
swarm-node-03            virtualbox   Running   tcp://192.168.99.102:2376   swarm-master
swarm-node-04            virtualbox   Running   tcp://192.168.99.103:2376   swarm-master
```
With the ```--debug``` flag turned on, I can get it to continue to fail on the same named VM, repeatedly with the tail end of the errors like:
```bash
generating server cert: /home/gstafford/.docker/machine/machines/swarm-node-01/server.pem ca-key=/home/gstafford/.docker/machine/certs/ca.pem private-key=/home/gstafford/.docker/machine/certs/ca-key.pem org=swarm-node-01
Using SSH client type: external
About to run SSH command:
sudo /etc/init.d/docker stop
&{/usr/bin/ssh [/usr/bin/ssh -o PasswordAuthentication=no -o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -i /home/gstafford/.docker/machine/machines/swarm-node-01/id_rsa -p 51309 docker@localhost sudo /etc/init.d/docker stop] []  <nil> <nil> <nil> [] <nil> <nil> <nil> <nil> false [] [] [] [] <nil>}
SSH cmd err, output: exit status 1: cat: can't open '/var/run/docker.pid': No such file or directory
sh: you need to specify whom to kill

SSH cmd error!
command: sudo /etc/init.d/docker stop
err    : exit status 1
output : cat: can't open '/var/run/docker.pid': No such file or directory
sh: you need to specify whom to kill
```

**Comment by nathanleclaire** [comment]
Hm, getting the same IP assigned to multiple VMs certainly is troubling.  So is the error message- I wonder if this is a race or simply a condition which we have forgotten to check (the daemon is up).

cc @tianon Have you been seeing any issues like this on the boot2docker side?  I know I've occasionally seen weirdness about starting / stopping using the init script.

In the meanwhile, @garystafford could I please get you to Gist the output of running a few `docker-machine create` commands (starting with the swarm master and then on to a few nodes) using the `-D` global flag for debug?  Thanks a lot!

**Comment by garystafford** [comment]
@nathanleclaire, here is a link to the first gist. I attempted to create my swarm node. It failed 6-7 times in a row with same issue: https://gist.github.com/garystafford/14ae413e488b26a81632.  

The same IP is typical when running a script that creates multiple nodes. If one fails and times out after 60 attempts, the next one builds and uses the existing failed nodes IP address. Depending on how many fail, the IPs can be used multiple times, etc.  

One thing I noticed and am trying again, I finally got the above swarm master node and all three subsequent nodes to build after stopping and starting networking. That has the effect of stopping all the VBox connections that were still live. I also restarted SSH, which didn't fix it.
